Automattic/wp-calypso
# Plugin installation on loading screen on a simple site that is on a business or e-commerce plan

#### Changes proposed in this Pull Request

### Finalised free plugin installation flow in a custom domained site
Completed free plugin installation flow, where a custom domain is purchased and set as the default domain. There are some known issues.
![freepluginflow](https://user-images.githubusercontent.com/3422709/120652980-e53fb800-c49d-11eb-808e-dd858d228188.gif)

### Created a site tester page with all the relevant urls for testing convenience

 `/marketplace/test/:site`


![image](https://user-images.githubusercontent.com/3422709/120653490-58492e80-c49e-11eb-94f5-0cdce77abed7.png)


### Known issues
- Right after the site transfer the default domain is changed to the wpcomstaging domain if the custom domain purchased is a recent purchase.
- Due to the above issue and this limbo state the selected site is not properly set.
- So the 2 buttons as well as the cross in the top left corner of the thank-you page does not work properly
- Plugin transfer failure states are not handled
- Already transferred site plugin installations do not work
   - Flow reproduce instructions 
      - Create a site
      - Set a custom domain as the primary domain
      - Install some plugin other than Yeost SEO to trigger an AT transfer
      - Go to url  `/marketplace/test/:site`
      - Click on button "Plugin details page"
      - Click on yoast free page
      - Loading screen should appear and plugin will be installed
      - After installation complete make sure thank-you page appears
      - Go to the wpcom home page and make sure the yoast seo plugin is installed

#### Testing instructions - Flow with untransferred site
1. Create a site
2. Set a custom domain as the primary domain
3. Go to url  `/marketplace/test/:site`
4. Click on button "Plugin details page"
5. Click on yoast free page
6. Loading screen should appear and plugin will be installed
7. After installation complete make sure thank-you page appears
8. Go to the wpcom home page and make sure the yoast seo plugin is installed
